app/models/user.py

```diff
@@ -8,6 +8,8 @@
 
 class Permission:
     GENERAL = 0x01
+    VENDOR = GENERAL | 0x02  # == 0x03 (will match GENERAL && VENDOR)
```

Comment:
Expand the binary. 

`GENERAL = 0b00000001`
`VENDOR = 0b00000001 OR 0b00000010 = 0b00000011`

Hence this notation (`GENERAL | 0x02`) forces the VENDOR permission to inherit all GENERAL permissions. Same for MERCHANT. Not sure if this is proper--alternative is to put `VENDOR | GENERAL` in the `permissions` field of the `Vendor` role--but in the end both methods result in the Vendor having a `permission` value of `0x03`.

It's really a question of if you want to think of the Vendor role having `GENERAL` and `VENDOR` permissions, or if you want to think of the `VENDOR` permission as inheriting all `GENERAL` permissions. It's written as the latter now, but I can change it if you like the sound of the former more.